app/controllers/heartbeat_controller.rb

```diff
@@ -0,0 +1,50 @@
+require "sidekiq/api"
+
+class HeartbeatController < ActionController::API
+  def ping
+    render body: "PONG"
+  end
+
+  def healthcheck
+    checks = {
+      database: database_alive?,
+      redis: redis_alive?,
+      sidekiq_processes: sidekiq_processes_checks,
+    }
+
+    status = checks.values.all? ? :ok : :service_unavailable
+
+    render status: status,
+           json: {
+             checks: checks,
+           }
+  end
+
+private
+
+  def database_alive?
+    ActiveRecord::Base.connection.active?
+  rescue PG::ConnectionBad
+    false
+  end
+
+  def redis_alive?
+    Sidekiq.redis_info
```

Comment:
I wonder if we could use `Redis#ping` like this:

```ruby
  Sidekiq.redis.ping
```

If `Sidekiq.redis` isn't a `Redis` object we might have to instantiate a new connection.